spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployer.java

```diff
@@ -326,6 +331,14 @@ private void validateStreamDeploymentRequest(StreamDeploymentRequest streamDeplo
 				.findById(streamName)
 				.orElseThrow(() -> new NoSuchStreamDefinitionException(streamDeploymentRequest.getStreamName()));
 
+		if ("kubernetes".equals(platformInfo.getPlatformType())) {
```

Comment:
I understand that the platform info is **only** available at the time of stream deployment and hence we can't get this fixed at the stream definition controller. Maybe let's discuss if we can restrict the stream name with the below regex for all the platform types - that way this validation could move up at the stream creation itself (via Stream Definition Controller)

Reply:
What's wrong with just making this a global rule and applying it to all platforms?

Reply:
I don't see any issue with that except some CF users might prefer the stream names with underscore in it ;)

Reply:
For consistency and to be able to support the UI (where this validation is really needed), I'd think we want to just apply this rule globally.